apps/src/code-studio/components/Libraries/LibraryCreationDialog.jsx

```diff
@@ -67,33 +90,85 @@ class LibraryCreationDialog extends React.Component {
     );
   };
 
+  validateInput = () => {
+    // Check if any of the checkboxes are checked
+    // If this changes the publishable state, update
+    let isChecked = $("input[type='checkbox']").is(':checked');
+    if (isChecked !== this.state.canPublish) {
+      this.setState({canPublish: isChecked});
+    }
+  };
+
   displayFunctions = () => {
     if (!this.state.loadingFinished) {
       return <div>Loading...</div>;
     }
+    let keyIndex = 0;
     return (
       <div>
-        <div>{this.state.libraryName}</div>
-        {this.state.selectedFunctionList.map(selectedFunction => {
-          let name = selectedFunction.functionName;
-          return <div key={name}>{name}</div>;
-        })}
+        <Heading2>
+          <b>Library Name: </b>
+          {this.state.libraryName}
+        </Heading2>
+        <form onSubmit={this.publish}>
+          <textarea
+            required
+            name="description"
+            rows="2"
+            cols="200"
+            style={styles.textarea}
+            placeholder="Write a description of your library"
+          />
+          {this.state.selectedFunctionList.map(selectedFunction => {
```

Comment:
When 'publish' is clicked, only the functions whose boxes are checked should be in "selectedFunctionsList."

Reply:
I didn't add this originally, but we'll probably need two lists of functions, the original functions parsed from the code, and the list of functions that have been checked.

Reply:
Whoops - forgot to only send the selected functions through. :woman_facepalming: Fixing.

I don't understand your second comment though. Can you elaborate?

Reply:
Haha it was meant to clarify the first comment, but sounds like it wasn't necessary (or helpful lol).

Reply:
For the createLibrary - I swapped out the full list of functions to be just the selected ones. Can you confirm that's the correct usage? (Latest commit)

Reply:
👍 Thank you!